metabase/metabase
# Alert isValid check only looks at configured channels

Closes https://github.com/metabase/metabase/issues/48406

### Description

Updates a couple components to functional, and leverages RTK hooks to grab formInput data.

### How to verify

1. Have an instance without Email configured.
2. Go to a question and open the create alert modal
3. Notice that even though Email is enabled, the "done" button is greyed out. This is because Email is not configured
4. The same should be true for any existing alert, and for Slack

### Demo
![image](https://github.com/user-attachments/assets/54942294-8e99-4694-9739-009828cad54e)

### Checklist
